server/src/main/java/org/opensearch/snapshots/RestoreService.java

```diff
@@ -629,6 +653,48 @@ private void checkAliasNameConflicts(Map<String, String> renamedIndices, Set<Str
                         }
                     }
 
+                    private String[] getIgnoreSettingsInternal() {
+                        // for non-remote store enabled domain, we will remove all the remote store
+                        // related index settings present in the snapshot.
+                        String[] indexSettingsToBeIgnored = new String[] {};
+                        if (false == RemoteStoreNodeAttribute.isRemoteStoreAttributePresent(clusterService.getSettings())) {
+                            indexSettingsToBeIgnored = ArrayUtils.concat(indexSettingsToBeIgnored, new String[] { "index.remote_store.*" });
+                        }
+                        return indexSettingsToBeIgnored;
+                    }
+
+                    private Settings getOverrideSettingsInternal() {
+                        final Settings.Builder settingsBuilder = Settings.builder();
+                        updateReplicationStrategy(
+                            settingsBuilder,
+                            request.indexSettings(),
+                            clusterService.getSettings(),
+                            clusterService.getClusterSettings()
+                        );
+                        // remote store settings needs to be overridden if the remote store feature is enabled in the
+                        // cluster where snapshot is being restored.
+                        MetadataCreateIndexService.updateRemoteStoreSettings(settingsBuilder, clusterService.getSettings());
+                        return settingsBuilder.build();
+                    }
+
+                    private void updateReplicationStrategy(
+                        Settings.Builder settingsBuilder,
+                        Settings requestSettings,
+                        Settings nodeSettings,
+                        ClusterSettings clusterSettings
+                    ) {
+                        // if remote store feature is enabled, override replication strategy to segment.
+                        // if `cluster.index.restrict.replication.type` is enabled and user have not provided
+                        // replication strategy setting with restore request, override replication strategy
+                        // setting to cluster default.
+                        if (isRemoteStoreAttributePresent(nodeSettings)) {
+                            settingsBuilder.put(SETTING_REPLICATION_TYPE, ReplicationType.SEGMENT);
+                        } else if (!INDEX_REPLICATION_TYPE_SETTING.exists(requestSettings)
```

Comment:
Will this if else ever lead to SEGMENT replication type? AFAIK, if only remote store attributes are present then only an index will be having SEGMENT replication type. The else block should always return DOCUMENT. Please correct me if otherwise.

Reply:
so we support both segment and docrep indices in the same cluster if remote store feature is not enabled. the idea here was if remote store is not enabled and user haven't provided replication strategy as request parameter, we will let user restore the index with the same replication setting that is present in snapshot, we don't need to override in such cases. however if user have `cluster.restrict.index.replication_type` settings enabled that means cluster can have only one type of replication strategy in that case we use the cluster level setting. more details here: #11669 .